tests/test_classifiers/test_sparse_sklearn_classifiers.py

```diff
@@ -0,0 +1,156 @@
+import copy
+from typing import Text, Type, Optional
+
+import pytest
+import scipy.sparse
+import numpy as np
+
+from rasa.shared.nlu.training_data.features import Features
+from rasa.shared.nlu.training_data.training_data import TrainingData
+from rasa.shared.nlu.training_data.message import Message
+from rasa.shared.nlu.constants import (
+    TEXT,
+    INTENT,
+    INTENT_NAME_KEY,
+    FEATURE_TYPE_SENTENCE,
+    PREDICTED_CONFIDENCE_KEY,
+)
+
+from rasa_nlu_examples.classifiers.sparse_naive_bayes_intent_classifier import (
+    SparseNaiveBayesIntentClassifier,
+)
+from rasa_nlu_examples.classifiers.sparse_logistic_regression_intent_classifier import (
+    SparseLogisticRegressionIntentClassifier,
+)
+from rasa_nlu_examples.classifiers.sparse_sklearn_intent_classifier import (
+    SparseSklearnIntentClassifier,
+)
+
+SPARSE_SKLEARN_INTENT_CLASSIFIERS = [
+    SparseNaiveBayesIntentClassifier,
+    SparseLogisticRegressionIntentClassifier,
+]
+
+
+def generate_dummy_training_data(
+    num_messages: int = 10,
+    with_dense_features: bool = False,
+    seed: Optional[int] = None,
+) -> TrainingData:
```

Comment:
Might it make sense to have an obvious example here with two intents that are easy to classify? It feels like the test might be stronger if we don't assume random data.

Reply:
Hm, but what does such a test really test. Wouldn't it just (only) test that a LR model is able to learn from a "simple dataset"? And do we really want to test that here? (Or in other words, do we want to test sklearn functionalities as well?)

Reply:
The main thing that comes to mind is our "intent_ranking" bit. Most of the time we only show the top 10 intents, which is something that we do on top of sklearn. I agree that we shouldn't test scikit-learn here, but since the ranking is something we add I could argue it's a fair check.

Reply:
We could maybe rewrite the sorting and actually... since when is predict supposed to return a tuple? 😅 I'll rewrite that and then we can either test the sorting or decide it's not test worthy anymore

Reply:
Good that we talked about this - there was actually a bug in the sorting (that was never triggered because no one tried to use that for n_samples > 1). I've separated the sorting now and added a test. Only todo left in there is a validation check for the schema of messages (i.e. did we add things to the message in the right way). Seems that doesn't work with the rasa version here yet. So I'd just leave that TODO in there as reminder for now.

Reply:
I had a hunch that there was a risk there, happy to hear the extra test fixed it 👍